mesh/mesh.cpp

```diff
@@ -4399,6 +4399,45 @@ Mesh::Mesh(real_t *vertices_, int num_vertices,
    FinalizeTopology();
 }
 
+Mesh::Mesh( const NURBSExtension& ext )
+ : attribute_sets(attributes), bdr_attribute_sets(bdr_attributes)
+{
+   SetEmpty();
+   /// make an internal copy of the NURBSExtension
+   NURBSext = new NURBSExtension( ext );
+
+   Dim              = NURBSext->Dimension();
+   NumOfVertices    = NURBSext->GetNV();
+   NumOfElements    = NURBSext->GetNE();
+   NumOfBdrElements = NURBSext->GetNBE();
+
+   NURBSext->GetElementTopo(elements);
+   NURBSext->GetBdrElementTopo(boundary);
+
+   vertices.SetSize(NumOfVertices);
+   if (NURBSext->HavePatches())
```

Comment:
It looks like the `else` case is not supported, so it should `MFEM_ABORT`?

Reply:
I can't tell at first glance whether the resulting `Mesh` would be invalid or simply trivial/empty (although I'm not sure what the practical use case for that would be).  Reviewing the code, it looks like checking mesh validity falls within the scope of `Mesh::Finalize()` (called by `Mesh::FinalizeMesh()` after the `if(){...}`), so I've left it without an `else{MFEM_ABORT(...)}` for now.

Reply:
It looks like `vertices` and `Nodes` are not set in the `else` case, and I'm not sure how that can possibly work or be useful. Unless there is a known example where the `else` case works, I think it is better to `MFEM_ABORT` here to inform the user where something went wrong, instead of failing later.

Reply:
The last commit I pushed added an `else` with `MFEM_ABORT`, since I couldn't think of any practical uses for the `else` case.